src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy

```diff
@@ -67,11 +71,33 @@ class IqPolicyEvaluatorWorkflowStep
     this.jobCredentialsId = jobCredentialsId
   }
 
+  @Override
+  String getApplicationId(){
+    if (applicationSelectTypePost == IqPolicyEvaluator.SELECT_APPLICATION_SELECT_TYPE) {
+      return listAppId
+    }
+    else {
+      return manualAppId
+    }
+  }
+
   @DataBoundConstructor
   IqPolicyEvaluatorWorkflowStep(final String iqStage,
-                                final String iqApplication) {
+                                final String applicationSelectTypePost,
```

Comment:
@DataBoundConstructor parameters are required by declarative pipelines. I envision a pipeline DSL where only applicationID or manualApplicationID are supplied and the workflow step figures the rest out, therefore @DataBoundSetter is preferred.

Reply:
parameters are required *fields* by declarative pipelines